File: 3.Presentation/ClientApp/app/models/onThisDay.ts
export interface IOnThisDay {
    id: number;
    subject: string;
    date: string;
    imageSource: string
}

export class OnThisDay implements IOnThisDay {
    constructor(
        public id: number,
        public subject: string,
        public date: string,
        public imageSource: string) { }
}

File: 0.Core/Entities/OnThisDay.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Core.Entities
{
    public class OnThisDay
    {   
        public int Id {get; set;}

        public string Subject { get; set; }

        public DateTime Date { get; set; }

        public string ImageSource { get; set; }

        public OnThisDay(string Subject, DateTime Date)
        {
            this.Subject = Subject;
            this.Date = Date;
        }

        public OnThisDay()
        {
            Subject = "";
            Date = DateTime.Now;
        }
    }
}
